src/usbg_common.c

```diff
@@ -23,6 +23,63 @@
 #include <unistd.h>
 #include <sys/sysmacros.h>
 
+int h2b(unsigned char c)
+{
+	/* check if char is a number */
+	if ((c >= '0') && (c <= '9'))
+		return c - '0';
+
+	/* check if char is upper case */
+	if ((c >= 'A') && (c <= 'F'))
+		return c - 'A' + 10;
+
+	/* check if char is lower case */
+	if ((c >= 'a') && (c <= 'f'))
+		return c - 'a' + 10;
+
+	return -EINVAL;
+}
+
+int usbg_write_guid(const char *path, const char *name,
+		      const char *file, const char *buf)
+{
+	int ret;
+	char guid[GUIDBINLEN];
+	char *tmp = guid;
+
+	if (strlen(buf) != GUIDCHARLEN)
+		return USBG_ERROR_INVALID_PARAM;
+
+	/* convert hex representation to binary */
+	for (int i = 0; i < GUIDBINLEN; i++) {
+		int hi, lo;
+
+		/* skip the four dashes */
+		if ((i == 4) || (i == 6) || (i == 8) || (i == 10))
+			buf++;
+
+		hi = h2b(*buf);
+		buf++;
+
+		lo = h2b(*buf);
+		buf++;
+
+		if (hi < 0 || lo < 0)
+			return USBG_ERROR_INVALID_PARAM;
+
+		if (hi > 15 || lo > 15)
+			return USBG_ERROR_INVALID_PARAM;
+
+		*tmp++ = (hi << 4) | lo;
```

Comment:
I would probably use a second counter for the output and directly write to `guid[j]` instead of using the `char *tmp`, because I consider it easier to grasp and debug. Both is fine, though.

Reply:
This is not valid anymore since we only use sscanf for parsing.